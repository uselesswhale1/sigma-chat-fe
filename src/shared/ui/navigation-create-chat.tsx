import { useState } from "react";
import { Box, Heading } from "@chakra-ui/react";
import { CreateChatForm } from "../models";
import { CreateChatModal } from "../../widgets";
import { PALLETE } from "../constants";

interface NavigationCreateChatProps {
  onCreate: (v: CreateChatForm) => void;
}

export const NavigationCreateChat = ({
  onCreate,
}: NavigationCreateChatProps): JSX.Element => {
  const [isOpen, setIsOpen] = useState<boolean>(false);

  const handleChatModalClose = (): void => {
    setIsOpen(false);
  };
  const handleChatModalOpen = (): void => {
    setIsOpen(true);
  };

  const handleChatCreate = (chat: CreateChatForm): void => {
    onCreate(chat);
    handleChatModalClose();
  };

  return (
    <>
      <Box
        data-testid="create-new-chat-test"
        border={"1px solid white"}
        cursor="pointer"
        onClick={handleChatModalOpen}
        my={"30%"}
        px="30%"
        py={5}
        borderRadius="10px"
      >
        <Heading size="xs" color={PALLETE.a} children="Create new chat" />
      </Box>

      {isOpen && (
        <CreateChatModal
          onCancel={handleChatModalClose}
          onCreate={handleChatCreate}
          open={isOpen}
        />
      )}
    </>
  );
};
