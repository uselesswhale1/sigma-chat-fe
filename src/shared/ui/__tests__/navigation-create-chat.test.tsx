import { render, screen } from "@testing-library/react";
import { NavigationCreateChat } from "../navigation-create-chat";
import userEvent from "@testing-library/user-event";

jest.mock("../../../widgets", () => ({
  CreateChatModal: () => <>CreateChatModal</>,
}));

describe("NavigationCreateChat should", () => {
  const props = {
    onCreate: jest.fn,
  };

  it("match snapshot", () => {
    const { container } = render(<NavigationCreateChat {...props} />);

    expect(container).toMatchSnapshot();
  });

  it("match snapshot with dialog", () => {
    const { container } = render(<NavigationCreateChat {...props} />);

    const btn = screen.getByTestId("create-new-chat-test");

    userEvent.click(btn);

    expect(container).toMatchSnapshot();
  });
});
