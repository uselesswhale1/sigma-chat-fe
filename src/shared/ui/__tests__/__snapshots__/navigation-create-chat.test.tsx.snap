// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationCreateChat should match snapshot 1`] = `
<div>
  <div
    class="css-1l5xc7t"
    data-testid="create-new-chat-test"
  >
    <h2
      class="chakra-heading css-1s27s82"
    >
      Create new chat
    </h2>
  </div>
</div>
`;

exports[`NavigationCreateChat should match snapshot with dialog 1`] = `
<div>
  <div
    class="css-1l5xc7t"
    data-testid="create-new-chat-test"
  >
    <h2
      class="chakra-heading css-1s27s82"
    >
      Create new chat
    </h2>
  </div>
</div>
`;
